src/main/scala/amba/ahb/RegisterRouter.scala

```diff
@@ -64,15 +76,49 @@ case class AHBRegisterNode(address: AddressSet, concurrency: Int = 0, beatBytes:
 
     out.ready := Bool(true)
     assert (d_phase || !out.valid)
+
+    genRegDescsJson(mapping:_*)
+    genOMRegMap(mapping:_*)
+  }
+    def genOMRegMap(mapping: RegField.Map*): OMRegisterMap = {
+    OMRegister.convert(mapping = mapping:_*)
+  }
+
+  def genRegDescsJson(mapping: RegField.Map*) {
```

Comment:
I'm hesitant to add these JSON files for APB/AHB/AXI4 as we were trying to move towards only using the object model (whose code you already added above). Are you planning on using this 'genRegDescJson'? If not, maybe better not to add it.